src/python/pants/backend/codegen/tasks/wire_gen.py

```diff
@@ -42,7 +40,23 @@ def register_options(cls, register):
   def __init__(self, *args, **kwargs):
     """Generates Java files from .proto files using the Wire protobuf compiler."""
     super(WireGen, self).__init__(*args, **kwargs)
-    self.java_out = os.path.join(self.workdir, 'gen-java')
+
+  @property
+  def synthetic_target_type(self):
+    return JavaLibrary
+
+  def is_gentarget(self, target):
+    return isinstance(target, JavaWireLibrary)
+
+  def sources_generated_by_target(self, target):
```

Comment:
You can use `self.add_to_build_file()` in a test to quickly add a build target. Is that what you mean?  I would rather test more explicitly for a few reasons:  

1) The code coverage numbers are reliable when you exercise code in unit tests. 
2) When something breaks, you can easily pin down and debug the problem in a unit test. It is often more difficult in an integration test.

(These opinions are mine and not necessarily shared throughout the pants community.)